pkgs/development/python-modules/rsyscall/default.nix

```diff
@@ -0,0 +1,70 @@
+{ lib
+, pythonPackages
+, pythonAtLeast
+, fetchPypi
+, librsyscall
+, nix
+, socat
+, pkg-config
+, openssh
+, coreutils
+}:
+
+pythonPackages.buildPythonPackage rec {
+  pname = "rsyscall";
+  version = "0.0.3";
+  disabled = ! pythonAtLeast "3.7";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "1wvviglibnkkh332f27hqzjyz7ipbp6h3p845h0x77y1vlsf4gxk";
+  };
+
+  buildInputs = [
+    pythonPackages.cffi
+    librsyscall
+  ];
+  nativeBuildInputs = [
+    pkg-config
+  ];
+  propagatedBuildInputs = with pythonPackages; [
+    trio
+    typeguard
+    pyroute2
+    outcome
+    nixdeps
+  ];
+  # rsyscall uses these to build containers on the fly
+  exportReferencesGraph = [
+    "nix" nix
+    "librsyscall" librsyscall
+    "openssh" openssh
+    "coreutils" coreutils
+  ];
+
+  checkInputs = with pythonPackages; [
+    pytest
+    pydocstyle
+    mypy
+    typing-extensions
+    socat
+  ];
+  # skipping tests because:
+  # ssh: the build user's login shell is /noshell
+  # net: /dev/net/tun doesn't exist
+  # nix: can't use Nix in the build sandbox
+  # pgid: /proc/sys/kernel/ns_last_pid doesn't exist
+  # fuse: /dev/fuse doesn't exist
+  checkPhase = ''
```

Comment:
Please use pytestCheckHook.

Reply:
pytestCheckHook has a bug/feature where it runs against the source tree rather than the installed version of the package. I'm not sure if that is intended - but it won't work here because there's a compiled Python extension in the installed version which is not in the source tree.

Reply:
could we fix this in preCheck? Otherwise please add a comment that the next person is not asking the same question again.